startup/90-functions.py

```diff
@@ -114,584 +248,854 @@ def print_det_keys(det_class):
+        flat_motor.move(motor_current+flat_motor_move,wait=True)
+        print([motor_current,flat_motor.position])
         beam_on()
-        set_detector(det,exposure_time=exposure_time,num_images=num_bright)
-        uid_bright = RE(count([det],num=1))[0]
+        RE(configure_area_det(det,acq_time=det_acq_time,exposure=det_exposure_dark))
+        uid_flat = RE(count([det]),md={'task':'linescan-flat',
+                                       'acq_time':det_acq_time,
+                                       'exposure':det_exposure_dark,
+                                       'flat_motor':flat_motor.name,
+                                       'flat_motor_move':flat_motor_move})[0]
         beam_off()
-        bright_taken = 'true'
+        flat_motor.move(motor_current,wait=True) 
     else:
-        uid_bright = 'none'
-        bright_taken = 'false'
-        
-        
-    if bright_taken == 'true':
-        if det.name == 'prosilica':
-            tiffs = get_tiff_list(hdr=db[-1])
-            t0 = fabio.open(tiffs[0]).data
-            img_bright = np.zeros((len(tiffs),t0.shape[0],t0.shape[1]))
-            for e,t in enumerate(tiffs):
-                img_bright[e,:,:] = fabio.open(tiffs[e]).data
-        else:
-            img_bright = np.array(list(db[-1].data('%s_image'%(det.name))))
+        uid_flat = 'none' 
 
-        if len(img_bright.shape) == 4:
-            img_bright = img_bright.mean(axis=1)
-            img_bright = img_bright.mean(axis=0)
-        if len(img_bright.shape) == 3:
-            img_bright = img_bright.mean(axis=0)
+    motor1_current_pos = motor1.position
+    print('>>> %s @ %.3f'%(motor1.name,motor1_current_pos))
+    motor2_current_pos = motor2.position
+    print('>>> %s @ %.3f'%(motor2.name,motor2_current_pos))
 
+    beam_on()
+    RE(configure_area_det(det,acq_time=det_acq_time,exposure=det_exposure_gridscan))
 
-        da_bright = xr.DataArray(data=img_bright.astype('float32'),
-                  coords=[np.arange(img_bright.shape[0]), np.arange(img_bright.shape[1])],
-                  dims=['pixel_y', 'pixel_x'],attrs=None
-                 )
-        ds['bright'] = da_bright
-        
+    uid_gridscan = RE(grid_scan([det],
+                                motor1,motor1_start,motor1_stop,motor1_nstep,
+                                motor2,motor2_start,motor2_stop,motor2_nstep,
+                                snake_axes=snake),
+                                md={'task':'gridscan','acq_time':det_acq_time,'exposure':det_exposure_gridscan})[0]
+    beam_off()
+
+    if come_back:
+        print('>>> Moving %s back to %.3f'%(motor1.name,motor1_current_pos))
+        motor1.move(motor1_current_pos)
+        print('>>> Moving %s back to %.3f'%(motor2.name,motor2_current_pos))
+        motor2.move(motor2_current_pos)    
+    
+    return uid_gridscan, uid_flat, uid_dark    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
```

Comment:
Those many blank lines are against the PEP8 standard. Would be great to run the code through `flake8` (for style check) and `black` (for automated formatting).

Reply:
I don't know what is flake8. Will try to figure out later.

Reply:
https://flake8.pycqa.org/en/latest/. We deploy as a part of our collection environments.